download/download_test.go

```diff
@@ -1,324 +1,483 @@
 package download
 
 import (
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"flag"
 	"fmt"
-	"io"
 	"net/http"
 	"net/http/httptest"
 	"os"
+	"path"
 	"path/filepath"
 	"testing"
+	"time"
 
+	createKey "github.com/NBISweden/sda-cli/create_key"
+	"github.com/golang-jwt/jwt"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
-type TestSuite struct {
+type DownloadTestSuite struct {
 	suite.Suite
-	accessToken string
+	tempDir        string
+	configFilePath string
+	accessToken    string
+
+	httpTestServer *httptest.Server
 }
 
-func (suite *TestSuite) createConfigFile(fileName, token string) os.File {
-	// Create conf file for sda-cli
-	confFile := fmt.Sprintf(`
-	access_token = %[1]s
-	host_base = inbox.dummy.org
-	encoding = UTF-8
-	host_bucket = inbox.dummy.org
-	multipart_chunk_size_mb = 50
-	secret_key = dummy
-	access_key = dummy
-	use_https = False
-	check_ssl_certificate = False
-	check_ssl_hostname = False
-	socket_timeout = 30
-	human_readable_sizes = True
-	guess_mime_type = True
-	encrypt = False
-	`, token)
+func (suite *DownloadTestSuite) SetupSuite() {
+	// Create a test httpTestServer
+	suite.httpTestServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		switch req.RequestURI {
+		case "/metadata/datasets/TES01/files":
+			// Set the response status code
+			w.WriteHeader(http.StatusOK)
+			// Set the response body
+			fmt.Fprint(w, `[
+            {
+                "fileId": "file1id",
+				"datasetId": "TES01",
+				"displayFileName": "file1.c4gh",
+                "filePath": "files/file1.c4gh",
+				"fileName": "4293c9a7-re60-46ac-b79a-40ddc0ddd1c6"
+            },
+			{
+                "fileId": "file2id",
+				"datasetId": "TES01",
+				"displayFileName": "file2.c4gh",
+                "filePath": "files/file2.c4gh",
+				"fileName": "4b40bd16-9eba-4992-af39-a7f824e612e2"
+            },
+			{
+                "fileId": "dummyFile",
+				"datasetId": "TES01",
+				"displayFileName": "dummy-file.txt.c4gh",
+                "filePath": "files/dummy-file.txt.c4gh",
+				"fileName": "4b40bd16-9eba-4992-af39-a7f824e612e1"
+            }
+        	]`)
+
+		case "/s3/TES01/files/dummy-file.txt.c4gh":
+
+			// Set the response status code
+			w.WriteHeader(http.StatusOK)
+
+			fmt.Fprint(w, "test content dummy file")
+
+		case "/s3/TES01/files/file1.c4gh":
+
+			// Set the response status code
+			w.WriteHeader(http.StatusOK)
+
+			fmt.Fprint(w, "test content file 1")
+		case "/s3/TES01/files/file2.c4gh":
+
+			// Set the response status code
+			w.WriteHeader(http.StatusOK)
+
+			fmt.Fprint(w, "test content file 2")
+		case "/metadata/datasets":
```

Comment:
```suggestion
		case "/s3/TES01/files/dummy-file.txt.c4gh":
			// Set the response status code
			w.WriteHeader(http.StatusOK)
			fmt.Fprint(w, "test content dummy file")
		case "/s3/TES01/files/file1.c4gh":
			// Set the response status code
			w.WriteHeader(http.StatusOK)
			fmt.Fprint(w, "test content file 1")
		case "/s3/TES01/files/file2.c4gh":
			// Set the response status code
			w.WriteHeader(http.StatusOK)
			fmt.Fprint(w, "test content file 2")
		case "/metadata/datasets":
```

Just some nitpicking, but try to keep whitespace usage consistent.